### PR TITLE
[CloudPC] Add connectivityStatus to CloudPc Entity

### PR DESCRIPTION
**Pull request by yayang3**
roadmap work item no: 14561

### CODE REVIEW

File: api-reference/beta/resources/cloudpc.md [hunk 1]
```diff
@@ -33,12 +33,14 @@ Represents a cloud-managed virtual desktop. This Cloud PC is also enrolled into
...
 |[Restore remote action](../api/manageddevice-restorecloudpc.md)|None|Restore a Cloud PC device to a previous state from a snapshot.|
 |[Bulk restore remote action](../api/manageddevice-bulkrestorecloudpc.md)|[cloudPcBulkRemoteActionResult](../resources/cloudpcbulkremoteactionresult.md)|Restore multiple Cloud PC devices with a single request that includes the IDs of Intune managed devices and a restore point date and time.|
+|[Get connectivity history](../api/cloudpc-getcloudpcconnectivityhistory.md)|[cloudPcConnectivityEvent](../resources/cloudpcconnectivityevent.md) collection|Get the Cloud PC connectivity history.|
 
 ## Properties
 
 |Property|Type|Description|
 |:---|:---|:---|
 |aadDeviceId|String|The Azure Active Directory (Azure AD) device ID of the Cloud PC.|
+|connectivityResult|[cloudPcConnectivityResult](../resources/cloudpcconnectivityresult.md)|The connectivity health check result of a cloud pc, including the updated timestamp and whether the cloud pc is able to be connected or not.|
```
Comment (RuiHou105) [inline api-reference/beta/resources/cloudpc.md:43]: Please update: cloud pc -> Cloud PC

File: api-reference/beta/resources/cloudpcconnectivityresult.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+|:---|:---|:---|
+|failedHealthCheckItems|[cloudPcHealthCheckItem](../resources/cloudpchealthcheckitem.md) collection|A list of failed health check items. If `status` is `available`, this property will be empty.|
+|status|[cloudPcConnectivityStatus](#cloudpcconnectivitystatus-values)|Connectivity status of the Cloud PC. Possible values are: `unknown`, `available`, `availableWithWarning`, `unavailable`.|
+|updatedDateTime|string|Datetime when the status was updated. The timestamp is shown in ISO 8601 format and Coordinated Universal Time (UTC). For example, midnight UTC on Jan 1, 2014 appears as `2014-01-01T00:00:00Z`. |
+
+### cloudPcConnectivityStatus values
+
+|Member|Value|Description|
+|:---|:---|:---|
+|unknown|0|The cloud pc connectivity status is unknown.|
```
Comment (RuiHou105) [inline api-reference/beta/resources/cloudpcconnectivityresult.md:29]: In the description of this file, you should describe as `Cloud PC`, not `cloud pc`. Please help correct :)

**Review by RuiHou105 — CHANGES_REQUESTED** [review]
Hi @yayang3 ,

1.In your PR, please correct the description from `cloud pc` to `Cloud PC`.

2.Please create a roadmap workitem and attach the work item number in this PR. Here is an [example]([Feature 14460](https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/14460): [DEVICES AND APPS][CLOUD PC] Add regionStatus to cloudPcSupportedRegion). If you don't have access, please do this firstly:
- Request access via [Improved ADO Access - 1ES Wikirequest to join](https://www.1eswiki.com/wiki/Improved_ADO_Access) .
- Join the Microsoft Graph Roadmap Users ([graphroadmap@microsoft.com](mailto:graphroadmap@microsoft.com)) security group on [idweb](https://idwebelements/GroupManagement.aspx?Group=graphroadmap&Operation=join) .

3.I see the build is failed. Not sure if your Github account has permissions. Please check. Thanks.

**Review by AshleyYangSZ — CHANGES_REQUESTED** [review]

File: api-reference/beta/api/cloudpc-getcloudpcconnectivityhistory.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+The following is an example of a request.
+
+<!-- {
+  "blockType": "request",
+  "name": "cloudpc.getCloudPcConnectivityHistory"
+}
+-->
+``` http
+GET https://graph.microsoft.com/beta/deviceManagement/virtualEndpoint/cloudPCs/9ec90ff8-fd63-4fb9-ab5a-aa4fdcc43ec9/getCloudPcConnectivityHistory
```
[the remark this answers is not in the document]
Comment (AshleyYangSZ) [reply api-reference/beta/api/cloudpc-getcloudpcconnectivityhistory.md:65]: And please make the build pass.

**Review by AshleyYangSZ — APPROVED** [review]

**Review by RuiHou105 — APPROVED** [review]

File: api-reference/beta/resources/cloudpcconnectivityevent.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+doc_type: resourcePageType
+---
+
+# cloudPcConnectivityEvent resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the details of Cloud PC connectivity event.
```
Comment (isvargasmsft) [inline api-reference/beta/resources/cloudpcconnectivityevent.md:16]: Would please explain a bit further what is a connectivity event?

File: api-reference/beta/resources/cloudpcconnectivityevent.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+### cloudPcConnectivityEventType values
+
+|Member|Value|Description|
+|:---|:---|:---|
+|unknown|0|The event type is unknown.|
+|userConnection|1|The event type is user connection.|
+|userTroubleshooting|2|The event type is user troubleshooting.|
+|deviceHealthCheck|3|The event type is device health check.|
+|unknownFutureValue|999|Evolvable enumeration sentinel value. Do not use.|
```
Comment (isvargasmsft) [inline api-reference/beta/resources/cloudpcconnectivityevent.md:44]: Would you be so kind to provide a bit more details on the description of each of these members, please? Just using the names does not clarify the scenarios each member covers.

File: api-reference/beta/resources/cloudpcconnectivityresult.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+doc_type: resourcePageType
+---
+
+# cloudPcConnectivityResult resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the details of the Cloud PC connectivity status.
```
Comment (isvargasmsft) [inline api-reference/beta/resources/cloudpcconnectivityresult.md:16]: Would you be so kind to explain a bit further what this resource is for? What represents a connectivity status and how is it related to the connectivity result?

File: api-reference/beta/api/cloudpc-getcloudpcconnectivityhistory.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+<!-- {
+  "blockType": "ignored"
+}
+-->
+
+``` http
+GET /deviceManagement/virtualEndpoint/cloudPCs/{id}/getCloudPcConnectivityHistory
+```
+
```
Comment (isvargasmsft) [inline api-reference/beta/api/cloudpc-getcloudpcconnectivityhistory.md:38]: Doesn't this function support any function parameters? If so, please include an H2 for Function parameters.

**Review by isvargasmsft — CHANGES_REQUESTED** [review]
Hello @yayang3, 

Very complete work in this PR, thank you very much. 

Would you be so kind to explain a bit more in the descriptions of the resources, please?

Thank you very much!

Isaac

File: api-reference/beta/api/cloudpc-getcloudpcconnectivityhistory.md (hunk 7 above)
Comment (yayang3) [reply api-reference/beta/api/cloudpc-getcloudpcconnectivityhistory.md:38]: no, it doesn't

File: api-reference/beta/resources/cloudpcconnectivityevent.md (hunk 4 above)
Comment (AshleyYangSZ) [reply api-reference/beta/resources/cloudpcconnectivityevent.md:16]: @yayang3 I think isvargasmsft hopes you could add the explain in the doc.

File: api-reference/beta/api/cloudpc-get.md [hunk 8]
```diff
@@ -226,6 +226,18 @@ Content-Type: application/json
...
+      "status": "unavailable",
+      "updatedDatetime": "2022-03-22T10:28:32.8260335Z",
+      "failedHealthCheckItems": [
+        {
+          "displayName": "DomainJoinCheck",
+          "result": "failure",
+          "lastHealthCheckDateTime": "2022-03-22T10:28:32.8260335Z",
+          "additionalDetails": "SessionHost unhealthy: SessionHost is not joined to a domain"
+        }
+      ]
```
[the remark this answers is not in the document]
Comment (isvargasmsft) [reply api-reference/beta/api/cloudpc-get.md:240]: @yayang3 this one is pending :)

[the remark this answers is not in the document]
Comment (yayang3) [reply api-reference/beta/api/cloudpc-get.md:240]: Sorry I missed this. user-list-cloudpcs operation won't return this property, so we don't need to update that:)

**Review by isvargasmsft — APPROVED** [review]

**Comment by RuiHou105** [comment]
Hi @Lauragra , this PR is ready for final review. Could you please help review/ approve it? Thanks ahead!

**Pull request merged**